### ISSUE TITLE
Poor performance on Pancreas dataset

### ISSUE THREAD

**Issue by yooceii** [milesial/Pytorch-UNet#175]
For some reason, the network is not able to find the Pancreas.
The config is:
INFO: Network:                                                                                                                                                                                                                                                        1 input channels                                                                                                                                                                                                                                              1 output channels (classes)                                                                                                                                                                                                                                   Bilinear upscaling                                                                                                                                                                                                                                    INFO: Starting training:                                                                                                                                                                                                                                              Epochs:          5                                                                                                                                                                                                                                            Batch size:      1                                                                                                                                                                                                                                            Learning rate:   0.1                                                                                                                                                                                                                                          Training size:   8315                                                                                                                                                                                                                                         Validation size: 923                                                                                                                                                                                                                                          Checkpoints:     True                                                                                                                                                                                                                                         Device:          cuda                                                                                                                                                                                                                                         Images scaling:  0.5

Here is an example of what it generated:
The original image:
![image](https://user-images.githubusercontent.com/19211761/81455443-3d9e6c00-915d-11ea-845f-8a8770f8e227.png)
The predict mask:
![image](https://user-images.githubusercontent.com/19211761/81455446-4000c600-915d-11ea-817c-845748cafbcd.png)
Label:
![image](https://user-images.githubusercontent.com/19211761/81455449-41ca8980-915d-11ea-9b5a-d75b071f72af.png)


Is it because of the area of Pancreas is small?

**Comment by talatccan** [comment]
Hi, I recommend you to try different learning rates like 0.0001 or 0.00001 can make difference. But for sure @milesial will answer better than me.

Also i have similliar problem about poor performans and i didnt want to open new issue.
Some of predictions and original masks are looking good but dice coeff changing arround 30%-40%. What can be wrong about it?

Training is continuing and epoch is 426/500 for now. What can i do for improve performance?
Train size: 742 and Val is10% of it.

Examples:
![resim](https://user-images.githubusercontent.com/10474206/81458522-7cefab80-91a3-11ea-83eb-b41f4474bc59.png)
![resim](https://user-images.githubusercontent.com/10474206/81458674-3f3f5280-91a4-11ea-9d43-dc0819e23440.png)

Dice Test:
![resim](https://user-images.githubusercontent.com/10474206/81458817-33a05b80-91a5-11ea-8aaa-0762b69cc873.png)
Train/Loss:
![resim](https://user-images.githubusercontent.com/10474206/81458833-503c9380-91a5-11ea-9861-b0caa91a2ed5.png)

**Comment by milesial** [comment]
@yooceii , the LR indeed seems high, as @talatccan said. Try a much lower one. Also check the loss curves to see if 5 epochs is enough, you may need more training.

It is really hard to diagnose problems with training on your different datasets, so I'm affraid I can't help you much. You have to play with different hyperparameters and optimizers, regularization, ... or just train longer and see what happens

**Comment by yooceii** [comment]
Looks like the batch size plays a major role here.
The performance improves a lot after changing to bigger batch size. 
For some reason, the code only worked with batch size 1 in previous experiments.
